### PR TITLE
Fix for writers with the same priority will get dequeued in no particular order

### PR DESCRIPTION
**Pull request by aaa2000**
@see https://github.com/portphp/portphp/issues/71

I did not succeed to implement a phpspec without add the `bossa/phpspec2-expect` dependency

### CODE REVIEW

**Comment by Baachi** [comment]
Hey @aaa2000 thanks for your work. I read the php bug report and as far as i undstood colder, we shouldn't rely on the `SplPriorityQueue`. Maybe we should adopt the behaviour from the [symfony event-dispatcher](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/EventDispatcher/EventDispatcher.php#L123-L127).

**Comment by aaa2000** [comment]
@Baachi I pushed a commit, replacing the `SplPriorityQueue`. Did you prefer this fix using an array ?

**Comment by sagikazarmark** [comment]
Not a huge fan of the expect thingy.

Maybe phpunit would be a better choice here.

File: src/StepAggregator.php [hunk 1]
```diff
@@ -195,4 +182,26 @@ private function buildPipeline()
...
+        foreach ($this->writers as $writer) {
+            $steps[-256][] = new Step\WriterStep($writer);
+        }
+
+        ksort($steps);
+        $stepsIterator = new \RecursiveIteratorIterator(
+            new \RecursiveArrayIterator($steps, \RecursiveArrayIterator:: CHILD_ARRAYS_ONLY)
+        );
+
+        return iterator_to_array($stepsIterator, false);
```
Comment (Baachi) [inline src/StepAggregator.php]: This looks strange, we have an array than we create a iterator with it and 2 lines later we transform the iterator back to an array.

**Review by Baachi — CHANGES_REQUESTED** [review]

File: src/StepAggregator.php (hunk 1 above)
Comment (aaa2000) [reply src/StepAggregator.php]: @Baachi I changed the implementation to use only array.

**Comment by aaa2000** [comment]
@sagikazarmark I removed the `bossa/phpspec2-expect` dependency using the technique described in the comment https://github.com/phpspec/prophecy/issues/130#issuecomment-64948054

**Comment by sagikazarmark** [comment]
Nice. (As far as I can see everzet suggested the same thing what I did, but I think it's ok for now)

**Comment by Baachi** [comment]
Can we merge this PR?

**Comment by aaa2000** [comment]
@Baachi on my side, it's ready

**Pull request merged**